togglevpn.sh

```diff
@@ -5,8 +5,37 @@ get_vpn_status() {
   scutil --nc status "$vpn" | head -n 1 | grep -i "connected"
 }
 
+disconnect_other_vpns() {
+  local current_vpn="$1"
+  
+  # Get all VPNs with their statuses in one call to scutil
+  scutil --nc list | grep "com.wireguard.macos" | while IFS= read -r line; do
+    # Extract VPN name from quotes (format: ... "VPN_NAME" ...)
+    vpn_name=$(echo "$line" | awk -F'"' '{print $2}')
+    
+    # Skip the current VPN that we're about to connect
+    if [[ "$vpn_name" == "$current_vpn" ]]; then
+      continue
+    fi
+    
+    # Only disconnect VPNs that are currently connected
+    if echo "$line" | grep -q "(Connected)"; then
+      scutil --nc stop "$vpn_name"
+	  sleep 1
+    fi
+  done
+}
+
 connect_vpn() {
   local vpn=$1
+  
+  # Check if we should turn off other VPNs
+  # Convert to lowercase for comparison (compatible with bash 3.2)
+  local env_value=$(echo "$is_turn_off_others" | tr '[:upper:]' '[:lower:]')
```

Comment:
I don't understand this. It seems you're referencing a variable that doesn't exist.